client/write.go

```diff
@@ -136,11 +139,11 @@ func (w *writeState) close() {
 }
 
 func (w *writeState) completionFn(result interface{}, err error) {
-	hostID := result.(topology.Host).ID()
+	hostID := result.(string)
 	// NB(bl) panic on invalid result, it indicates a bug in the code
 
-	w.Lock()
-	w.pending--
+	pending := atomic.AddInt32(&w.pending, -1)
+	var addToSuccess int32
 
 	if err != nil {
 		w.errors = append(w.errors, fmt.Errorf("error writing to host %s: %v", hostID, err))
```

Comment:
Do we need to protect `w.errors` here in case multiple go routines access it at the same time?